frontend/packages/console-shared/src/utils/operator-utils.ts

```diff
@@ -0,0 +1,52 @@
+import * as _ from 'lodash';
+import {
+  K8sResourceKind,
+  HorizontalPodAutoscalerKind,
+  K8sKind,
+  referenceForModel,
+} from '@console/internal/module/k8s';
+import { ClusterServiceVersionKind } from '@console/operator-lifecycle-manager/src';
+import { CAMEL_SOURCE_INTEGRATION } from '@console/knative-plugin';
+import { TopologyResources, EventSourcesData } from '../types';
+import { getOperatorBackedServiceKindMap, getKnativeDynamicResources } from './resource-utils';
+
+export type RelatedResources = {
+  hpas?: HorizontalPodAutoscalerKind[];
+};
+
+const eventSourceData: EventSourcesData = {
+  loaded: false,
+  eventSourceModels: [],
+  eventSourceChannels: [],
+};
+
+export const isOperatorBackedKnResource = (obj: K8sResourceKind, resources: TopologyResources) => {
+  const eventSourceProps = eventSourceData.eventSourceModels.map((model: K8sKind) =>
+    referenceForModel(model),
+  );
+  return !!_.find(getKnativeDynamicResources(resources, eventSourceProps), (evsrc) =>
+    obj.metadata?.labels?.[CAMEL_SOURCE_INTEGRATION]?.startsWith(evsrc.metadata.name),
+  );
+};
```

Comment:
This doesn't make any sense, your const `eventSourceData` has an empty array of modals and channel then your eventSourceProps will be empty as well. I think you want to use the `getDynamicEventSourcesModelRefs` from fetch-dynamic-eventsource-utils.ts

Reply:
fixed it